big_tests/tests/mongoose_helper.erl

```diff
@@ -341,30 +341,40 @@ wait_until(Fun, ExpectedValue) ->
     wait_until(Fun, ExpectedValue, #{}).
 
 %% Example: wait_until(fun () -> ... end, SomeVal, #{time_left => timer:seconds(2)})
-wait_until(Fun, ExpectedValue, Opts) ->
-    Defaults = #{validator => fun(NewValue) -> ExpectedValue =:= NewValue end,
-                 expected_value => ExpectedValue,
-                 time_left => timer:seconds(5),
+%% if expected value is a function with arity 1, it's treated as a validation function.
+wait_until(Fun, ExpectedValue, Opts0) ->
+    Defaults = #{time_left => timer:seconds(5),
                  sleep_time => 100,
-                 history => [],
                  name => timeout},
-    do_wait_until(Fun, maps:merge(Defaults, Opts)).
-
-do_wait_until(_Fun, #{expected_value := ExpectedValue,
-                      time_left := TimeLeft,
-                      history := History,
-                      name := Name} = Opts) when TimeLeft =< 0 ->
+    Opts1 = maps:merge(Defaults, Opts0),
+    TimeLeft = maps:get(time_left, Opts1),
+    Opts = Opts1#{waiting_time => 0, max_waiting_time => TimeLeft,
```

Comment:
why the max_waiting_time is added to opts? I cannot see anywhere in the code the usage of it.

Reply:
For the traceability reasons only.

Reply:
it's nice to have an unchanging initial value among the dynamically changing counter

Reply:
Ok, now I understand. Thanks for explaining it to me.